sandplover/plan.py

```diff
@@ -1270,84 +1326,174 @@ def compute_shoreline_roughness(shore_mask, land_mask, **kwargs):
 
         And now, we can proceed with the calculation.
 
-        Compute roughnesses
-
-        >>> from sandplover.plan import compute_shoreline_roughness
-
-        >>> rgh0 = compute_shoreline_roughness(sm0, lm0)
-        >>> rgh1 = compute_shoreline_roughness(sm1, lm1)
+    .. plot::
+        :include-source:
+        :context: close-figs
 
-        Make the plot
+        >>> # Compute roughnesses
+        >>> from sandplover.plan import compute_shoreline_roughness_area
+        >>> rgh0 = compute_shoreline_roughness_area(sm0, lm0)
+        >>> rgh1 = compute_shoreline_roughness_area(sm1, lm1)
 
+        >>> # Make the plot
         >>> fig, ax = plt.subplots(1, 2, figsize=(6, 3))
         >>> golf.quick_show("eta", idx=15, ax=ax[0])
         >>> _ = ax[0].set_title("roughness = {:.2f}".format(rgh0))
         >>> golf.quick_show("eta", idx=-1, ax=ax[1])
         >>> _ = ax[1].set_title("roughness = {:.2f}".format(rgh1))
+
+    .. seealso::
+
+        See this :doc:`example using this metric
+        </guides/examples/computations/shoreline_roughness_perfect_direct>`
+        and comparing it to the theoretical value for a perfect half-circle
+        delta
+
+    .. seealso::
+
+        See :doc:`this example
+        </guides/examples/computations/comparing_shoreline_roughness_metrics>`,
+        which compares all of the "shoreline roughness" metrics implemented
+        in sandplover.
+
     """
-    # extract data from masks
-    if isinstance(land_mask, LandMask):
-        land_mask = land_mask.mask
-        _lm = land_mask.values
-        _dx = float(land_mask[land_mask.dims[0]][1] - land_mask[land_mask.dims[0]][0])
-    elif isinstance(land_mask, xr.core.dataarray.DataArray):
-        _lm = land_mask.values
-        _dx = float(land_mask[land_mask.dims[0]][1] - land_mask[land_mask.dims[0]][0])
-    elif isinstance(land_mask, np.ndarray):
-        _lm = land_mask
+    # process the shore_mask, stripping down to array coordinates regardless of
+    # the input. This simplifies the roughness calculation, because we know
+    # we are dealing with array coords always.
+    if isinstance(shore_mask, ShorelineMask):
+        shore_mask = shore_mask.mask
+        _sm = shore_mask.values
+        _dx = float(
+            shore_mask[shore_mask.dims[0]][1] - shore_mask[shore_mask.dims[0]][0]
+        )
+    elif isinstance(shore_mask, xr.core.dataarray.DataArray):
+        _sm = shore_mask.values
+        _dx = float(
+            shore_mask[shore_mask.dims[0]][1] - shore_mask[shore_mask.dims[0]][0]
+        )
+    elif isinstance(shore_mask, np.ndarray):
+        _sm = shore_mask
         _dx = 1
     else:
-        raise TypeError(f"Invalid type {type(land_mask)}")
+        raise TypeError(f"Invalid type {type(shore_mask)}")
```

Comment:
This little block of code gets copy-pasted around a few times, should it be its own utility?